tests/virt_autotest/install_package.pm

```diff
@@ -102,6 +102,8 @@ sub install_package {
     ###Install required package for window guest installation on xen host
     if (get_var('GUEST_LIST') =~ /^win-.*/ && (is_xen_host)) { zypper_call '--no-refresh --no-gpg-checks in mkisofs' }
 
+    #Subscribing packagehub from SLE 15-SP4 onwards that enables access to many useful software tools
+    virt_autotest::utils::subscribe_extensions_and_modules(reg_exts => 'PackageHub') if (is_sle('>=15-sp4') and !is_s390x);
```

Comment:
Please be aware that packages in PackageHub is not supported by SuSE, which means that if there are bugs on these tools, we can't open bugs to let dev fix. If we rely on these tools for testing, it needs to be considered how to proceed when the tools turn problematic.  So I'd suggest to use supported tools instead, although sometimes it may be not that convenient. But this function subscribe_extensions_and_modules is pretty useful.

Reply:
This is the last resort for someone to perform test and at the same time facilitate test run. Our focus is testing 'Virtualizaiton', so no worry.  We do not do thorough testing on tools, for exmaple, sshpass. We just use the very basic functionality of tools themselves, so no worry about bugs from them or even report them, basically you won't see them. But it is really convenient if a certain tool can be employed to facilitate our test runs. By the way, if you want openQA packages on SLES, you have to have PackageHub.

Reply:
Yeah, that's true mostly. Then let's keep this bottom principle, that although packagehub is enabled, please pick supported tools first if existing, otherwise if we have to use packagehub tools, keep usage as basic as possible so as to avoid potential issues :).